### PR TITLE
Add MediaItemTranslation and make it editable

### LINKED ISSUE #2469: Media Library: Add option to add description to every MediaGroupMediaItem
## Type

- Feature

## Problem description

When creating a media gallery, you can add images.
Sometimes you feel the need to have the option to add a description to every image.

Example:
Title + Description + Link to the specific page.
schermafbeelding 2018-02-16 om 11 02 28

## Expected behavior / Proposed solutions

We should add a button to every image, which opens a pop-up, where we can change the "text".
schermafbeelding 2018-02-16 om 11 03 54

### PR DESCRIPTION
**Pull request by jeroendesloovere**
## Type

- Feature

## Resolves the following issues

Fixes #2469 
This PR handles the Backend part.

## Pull request description

MediaItem has now more descriptive fields and they can be filled in for every language:
* Title
* Caption
* CaptionLink
* Alt text
* Description
schermafbeelding 2018-02-21 om 13 22 03

schermafbeelding 2018-02-21 om 13 43 48

### CODE REVIEW

**Comment by jeroendesloovere** [comment]
@forkcms/core-contributors 

I would like some advice about what you guys find the best way of integrating getTranslation() in back- and frontend. So we can access mediaItem.getTranslation().getTitle()/getCaption()/getDescription()/getAlt()/...

File: src/Backend/Modules/MediaLibrary/Actions/MediaItemEdit.php [hunk 1]
```diff
@@ -75,12 +77,12 @@ public function execute(): void
             $this->getBackLink(
                 [
                     'report' => 'media-item-edited',
-                    'var' => $updateMediaItem->title,
+                    'var' => $updateMediaItem->url,
```
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Actions/MediaItemEdit.php]: why did you change this?

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php [hunk 2]
```diff
@@ -235,21 +258,28 @@ public static function createFromMovieUrl(
     public static function fromDataTransferObject(MediaItemDataTransferObject $mediaItemDataTransferObject): ?MediaItem
     {
         if (!$mediaItemDataTransferObject->hasExistingMediaItem()) {
-            throw new \BadFunctionCallException('This method can not be used to create a new media item');
+            throw new BadFunctionCallException('This method can not be used to create a new media item');
```
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: Exception thrown if a callback refers to an undefined function or if some arguments are missing. 

I think a RuntimeException would be better here

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php [hunk 3]
```diff
@@ -235,21 +258,28 @@ public static function createFromMovieUrl(
...
+
+            $mediaItem->addTranslation(MediaItemTranslation::fromDataTransferObject($mediaItemTranslationDataTransferObject));
+        }
+
         return $mediaItem;
     }
 
     public function jsonSerialize(): array
     {
+        $locale = BackendLocale::workingLocale();
```
Comment (carakas) [inline src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: not a fan of this, I think a method to set the active language is better with a fallback like this for instance, but now if you would try to use this in the frontend it would fail

Comment (carakas) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: upi can then also have a method, getActiveTranslation

File: src/Backend/Modules/MediaLibrary/Actions/MediaItemEdit.php (hunk 1 above)
Comment (jeroendesloovere) [reply src/Backend/Modules/MediaLibrary/Actions/MediaItemEdit.php]: Before
MediaItem->getTitle();

After
MediaItem->getTranslation(locale)->getTitle();

That is why I'm now using $updateMediaItem->url, because ->title doesn't exist anymore

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php (hunk 2 above)
Comment (jeroendesloovere) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: But,... some arguments are missing? That is why I throw error.

Comment (jeroendesloovere) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: Fixed. Changed to RuntimeException as requested.

File: src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php (hunk 3 above)
Comment (jeroendesloovere) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: Can you tell me how do you like to see Locale (Backend/Frontend) integrated?

Comment (jeroendesloovere) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: @carakas, any example?

Comment (jeroendesloovere) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: poke @carakas

Comment (carakas) [reply src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php]: just add a property for the current locale that isn't persisted to the entity, then add a method to set/change that locale and use that to get the correct translation

don't know the location of a code snippet by heart
